3rd-party/treematch/k-partitioning.c

```diff
@@ -1,22 +1,26 @@
+/*
+ * Copyright (c) 2022      IBM Corporation.  All rights reserved.
```

Comment:
It seems weird to have IBM as the sole copyright holder in many of these files. There was an original copyright from INRIA (https://github.com/open-mpi/ompi/pull/10863/files#diff-22612df7e140efa8b8cffcfe699b49096a271483f85391fce5aeff1ceb977a31L1) in some files. I guess that would apply to the other files too? Maybe it should be added to all treematch files as well?

Reply:
I deleted the IBM copyrights, since this is just renaming a bunch of symbols. I don't think I should add INRIA copyrights since I'm not sure INRIA is the original author.